pkg/cli/importimage/importimage.go

```diff
@@ -187,7 +188,14 @@ func (o *ImportImageOptions) Run() error {
 		return err
 	}
 
-	result, err := o.imageClient.ImageStreamImports(isi.Namespace).Create(context.TODO(), isi, metav1.CreateOptions{})
+	result := &imagev1.ImageStreamImport{}
+	err = o.imageClient.RESTClient().Post().
+		Namespace(isi.Namespace).
+		Resource(imagev1.Resource("imagestreamimports").Resource).
+		Body(isi).
+		Timeout(time.Hour).
```

Comment:
For one increasing this to 1h is not acceptable? Two, have you tried `--request-timeout`?

Reply:
I liked your idea of having this as a command line flag (I could not locate it on the current version, at least). What do you think @dmage ?

Reply:
We are fixing regression, --request-timeout (if it works) may be considered as a workaround, but it used to work without additional flags. 1h is acceptable for everybody (who has supported configuration), we have the same timeout in [openshift-controller-manager](https://github.com/openshift/openshift-controller-manager/blame/e1af3a50434542be4e9fd227e5203fda41835b06/pkg/image/controller/imagestream_controller.go#L310).

Reply:
Ah, yes, I was not aware of `oc options`. Let me see if that works.

Reply:
Wow, thanks @soltysh 

```
$ /tmp/oc --request-timeout=1h import-image myruby --from=foo --all --confirm --insecure --loglevel=8
...
I0118 16:17:48.882223  101857 round_trippers.go:422] POST https://api.crc.testing:6443/apis/image.openshift.io/v1/namespaces/rmarasch/imagestreamimports?timeout=1h0m0s
...

I am inclined to just close the BZ now. What do you reckon @dmage ?
```

Reply:
@soltysh is there a way to change the default value for `oc import-image`?

Reply:
It's a field on `rest.Config` so it's doable, but then again the default is no time-out, and it's the server that times out the request unless explicitly asked not to do it. I'd personally update help and docs with information about `--request-timeout` it's there, just not front of people's eyes.

Reply:
The default was no time-out, but today it's 34 seconds:

https://github.com/kubernetes/kubernetes/blob/f35df2f8411ea68891c9fa829733c497002d4b97/staging/src/k8s.io/apiserver/pkg/endpoints/handlers/rest.go#L499

https://github.com/kubernetes/kubernetes/blob/018942a92a90f91261ba63971827f2ec6b64587e/staging/src/k8s.io/apiserver/pkg/endpoints/handlers/create.go#L61-L79

Therefore `oc import-image` stopped to work for big repositories. But given that customers didn't report about the problem and that the API server already has a bit incompatible change, maybe documenting this is sufficient.